run.c

```diff
@@ -361,13 +369,17 @@ int argmax(float* v, int n) {
 
 long time_in_ms() {
     struct timespec time;
+#ifndef _WIN32
```

Comment:
should this be moved up one row? as `time` is not used in the windows version?

Reply:
Yea, originally I was using my previous timer definition on windows which used timespec struct, but changed to GetTickCount since that is standard pattern on windows, can move the ifdef

Reply:
I can use the approach from #96 to overload the clock_gettime definition in a win.h if you want me to clean up the timer related ifdef

Reply:
Yeah, we could see what that looks like. I don't really understand how the overloading happens though. By including the file up top, the function down below gets overwritten?

Reply:
I refactored and tested but it turns out that mingw doesn't have a timespec_get implementation so that would disallow cross-compilation. The best windows time_in_ms solution for time deltas is the ifdef I proposed here with GetTickCount()

Reply:
I'm not sure I understand. Isn't the timespec_get only used in the non-Windows version? So it should be moved inside the ifndef? It's spurious in Windows because we just call GetTickCount

Reply:
On linux the function is clock_gettime(). My original patch proposed using timepsec_get because it exists on linux and MSVC and matches the same function prototype as clock_gettime(), and this was before I tried the cross-compile option which I think is convenient.

In theory the implementation could be hidden in a header that is only included on windows to implement clock_gettime() by using timespec_get() underneath. 

But that breaks mingw compiles, so we should stick with GetTickCount(), which does not return seconds+nano, so it needs to be placed in ifdefs in the time_in_ms() function since it returns ms resolution timing.